server/src/test/java/org/apache/calcite/avatica/ha/ConnectionPropertiesHATest.java

```diff
@@ -0,0 +1,227 @@
+    String url = SERVERS.getJdbcUrl(START_PORT, Driver.Serialization.PROTOBUF);
+
+    AvaticaConnection conn = (AvaticaConnection) DriverManager.getConnection(url, properties);
+    String uri1 = getConnectionURI(conn);
+    Assert.assertEquals(urls[0], uri1);
+  }
+
+  @Test
+  public void testConnectionPropertiesHATestInvalidLBUrl() throws Exception {
+    Properties properties = new Properties();
+    properties.put(BuiltInConnectionProperty.USE_CLIENT_SIDE_LB.name(), "true");
+    StringBuilder sb = new StringBuilder();
+    for (int i = 0; i < 3; i++) {
+      sb.append("http://invalid:").append(START_PORT + i).append(",");
+    }
+    properties.put(BuiltInConnectionProperty.LB_URLS.name(), sb.substring(0, sb.length() - 1));
+    properties.put(BuiltInConnectionProperty.LB_STRATEGY.name(),
+        RoundRobinLBStrategy.class.getName());
+    String url = SERVERS.getJdbcUrl(START_PORT, Driver.Serialization.PROTOBUF);
+    try {
+      DriverManager.getConnection(url, properties);
+    } catch (RuntimeException re) {
+      Assert.assertTrue(re.getCause() instanceof UnknownHostException);
+    }
+  }
+
+  @Test
+  public void testConnectionPropertiesHALBFailover() throws Exception {
+    Properties properties = new Properties();
+    properties.put(BuiltInConnectionProperty.USE_CLIENT_SIDE_LB.name(), "true");
+    properties.put(BuiltInConnectionProperty.LB_CONNECTION_FAILOVER_RETRIES.name(), "3");
+    properties.put(BuiltInConnectionProperty.HTTP_CONNECTION_TIMEOUT.name(), "300");
+    String lbUrls = "http://localhost:" + START_PORT;
+    properties.put(BuiltInConnectionProperty.LB_URLS.name(), lbUrls);
+    properties.put(BuiltInConnectionProperty.LB_STRATEGY.name(),
+        RoundRobinLBStrategy.class.getName());
+    String url = SERVERS.getJdbcUrl(START_PORT, Driver.Serialization.PROTOBUF);
+    try {
+      DriverManager.getConnection(url, properties);
+    } catch (RuntimeException re) {
+      Assert.assertTrue(re.getCause() instanceof UnknownHostException);
+    }
+  }
+
+  @Test
+  public void testConnectionPropertiesHAHttpConnectionTimeout5Sec() throws Exception {
+    Properties properties = new Properties();
+    properties.put(BuiltInConnectionProperty.USE_CLIENT_SIDE_LB.name(), "true");
+    properties.put(BuiltInConnectionProperty.HTTP_CONNECTION_TIMEOUT.name(), "5000");
+    properties.put(BuiltInConnectionProperty.LB_CONNECTION_FAILOVER_RETRIES.name(), "0");
+    String urls = "http://240.0.0.1:" + START_PORT;
+    properties.put(BuiltInConnectionProperty.LB_URLS.name(), urls);
+
+    String url = SERVERS.getJdbcUrl(START_PORT, Driver.Serialization.PROTOBUF);
+    long startTime = System.currentTimeMillis();
+    try {
+      DriverManager.getConnection(url, properties);
+    } catch (RuntimeException re) {
+      long endTime = System.currentTimeMillis();
+      long elapsedTime = endTime - startTime;
+      Assert.assertTrue(elapsedTime < Timeout.ofMinutes(3).toMicroseconds());
```

Comment:
Are you comparing milli and micro seconds here ?

Reply:
Done. Changed it to milliseconds.